src/Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -863,16 +863,16 @@ public AccountHistory GetHistory(HdAccount account)
         }
 
         /// <inheritdoc />
-        public IEnumerable<AccountBalance> GetBalances(string walletName, string accountName = null)
+        public IEnumerable<AccountBalance> GetBalances(string walletName, string accountName = null, int confirmations = 0)
         {
             lock (this.lockObject)
             {
-                int tipHeight = this.GetWallet(walletName).AccountsRoot.First().LastBlockSyncedHeight ?? 0;
```

Comment:
This is a change in behavior, not saying its wrong but it means a wallet that is syncing will use the tip of the chain instead to tip of the wallet.

Reply:
Any suggestions as to what would be the "most" correct is appreciated. Whatever we decide should be implemented uniformly insofar as wallet/account/address balances are concerned.

Reply:
probably this could be a problem if there is a reorg because I think that in that case we may pick balances even of rewinded blocks (if they are not removed from wallet of course, if they are then I don't see a problem even if would be safe to stay on wallet tip)

Reply:
The reorged trx should be removed (and added later if they are on the connected blocks)

As for this logic I am not sure, it probably means transactions will have more confirmations form point of view of the wallet.
The wallet should probably no be used if its tip is considered in IBD (but I don't think we enforce that)

Reply:
@quantumagi i'd say leave it as you did to save a round trip to disk, doesn't seem worth it (unless you have the wallet height cached).